scale/scale_test.go

```
@@ -0,0 +1,113 @@
+	return m.client.Deployment, nil
+}
+
+func (m *MockDeployment) List(opts api.ListOptions) (*extensions.DeploymentList, error) {
+	return nil, nil
+}
+
+func (m *MockDeployment) Delete(name string, options *api.DeleteOptions) error {
+	return nil
+}
+
+func (m *MockDeployment) Create(*extensions.Deployment) (*extensions.Deployment, error) {
+	return nil, nil
+}
+
+func (m *MockDeployment) UpdateStatus(*extensions.Deployment) (*extensions.Deployment, error) {
+	return nil, nil
+}
+
+func (m *MockDeployment) Watch(opts api.ListOptions) (watch.Interface, error) {
+	return nil, nil
+}
+
+func (m *MockDeployment) Rollback(*extensions.DeploymentRollback) error {
+	return nil
+}
+
+func (m *MockKubeClient) Deployments(namespace string) kclient.DeploymentInterface {
+	return &MockDeployment{
+		client: m,
+	}
+}
+
+func NewMockKubeClient() *MockKubeClient {
+	return &MockKubeClient{
+		Deployment: &extensions.Deployment{
+			Spec: extensions.DeploymentSpec{
+				Replicas: 3,
+			},
+		},
+	}
+}
+
+func NewMockPodAutoScaler(kubernetesDeploymentName string, kubernetesNamespace string, max int, min int) *PodAutoScaler {
+	mockClient := NewMockKubeClient()
+
+	return &PodAutoScaler{
+		Client:     mockClient,
+		Min:        min,
+		Max:        max,
+		Deployment: kubernetesDeploymentName,
+		Namespace:  kubernetesNamespace,
+	}
+}
+
+func TestScaleUp(t *testing.T) {
+	p := NewMockPodAutoScaler("test", "test", 5, 1)
+
+	_ = p.ScaleUp()
+	deployment, _ := p.Client.Deployments("test").Get("test")
+	assert.Equal(t, int32(4), deployment.Spec.Replicas)
+	_ = p.ScaleUp()
+	deployment, _ = p.Client.Deployments("test").Get("test")
+	assert.Equal(t, int32(5), deployment.Spec.Replicas)
+
+	err := p.ScaleUp()
+	assert.NotNil(t, err)
+	deployment, _ = p.Client.Deployments("test").Get("test")
+	assert.Equal(t, int32(5), deployment.Spec.Replicas)
+}
+
+func TestScaleDown(t *testing.T) {
+	p := NewMockPodAutoScaler("test", "test", 5, 1)
+
+	_ = p.ScaleDown()
```

again, discarding error smell